src/core/connection.c

```diff
@@ -5521,10 +5521,32 @@ QuicConnRecvDatagramBatch(
         CxPlatZeroMemory(HpMask, BatchCount * CXPLAT_HP_SAMPLE_LENGTH);
     }
 
+    uint8_t prevPacket = 0;
```

Comment:
nit: casing:
```suggestion
    uint8_t PrevPacket = 0;
```
Though, I wonder if we should be changing something more fundamental and these shouldn't be considered part of the same batch to begin with.

Reply:
Removed PrevPacket.
I feel grouping same KeyType packets may complicate this further.

Reply:
> I feel grouping same KeyType packets may complicate this further.

Perhaps, but I still think it's likely a better design.

Reply:
Please check whether current changes are good. I am avoiding batching of packets if KeyType changes.